src/components/resultsbox/calculator.js

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { DrugList } from "../druglist";
+import GridTable from "./gridtable";
+
+const determineMinDose = (weightInput, minDosePerKg, maxDailyDose, freq) => {
+  const minDosePerDay =
+    weightInput * minDosePerKg >= maxDailyDose
+      ? maxDailyDose
+      : weightInput * minDosePerKg;
+  const minDose = divideDoseByFreq(freq, minDosePerDay);
+  return minDose;
+};
+
+const determineMaxDose = (weightInput, maxDosePerKg, maxDailyDose, freq) => {
+  const maxDosePerDay =
+    weightInput * maxDosePerKg >= maxDailyDose
+      ? maxDailyDose
+      : weightInput * maxDosePerKg;
+  const maxDose = divideDoseByFreq(freq, maxDosePerDay);
+  return maxDose;
+};
+
+const divideDoseByFreq = (freq, dose) => {
+  let dividedDose = 0;
+  if (freq === "BD") {
+    dividedDose = (dose / 2).toFixed(2);
+  } else if (freq === "TDS") {
+    dividedDose = (dose / 3).toFixed(2);
+  } else if (freq === "QDS") {
+    dividedDose = (dose / 4).toFixed(2);
+  }
```

Comment:
This is a classic scenario for a switch statement. 

```suggestion
  switch(freq) {
      case "BD": return (dose / 2).toFixed(2);
      case "TDS": return (dose / 3).toFixed(2)
      case "QDS": return (dose / 4).toFixed(2)
      default: return 0;
  }
```

Reply:
Also, it will be impossible to tell what those acronyms mean. Spelling them out somewhere might make more sense.
Maybe an object for static values like those?
```
const FREQUENCIES = {
    BONE_DANCER: "BD",
    TOMORROWS_DAWN_SENSATION: "TDS",
    QUARTERLY_DEUTSCH_SAUSAGES: "QDS"
}
```
then reference as needed. No problem with typos and such